src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -461,24 +461,30 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
             PropertyInfo setter = dictionaryType.GetProperty("Item", DeclaredOnlyLookup)!;
             foreach (IConfigurationSection child in config.GetChildren())
             {
-                object? item = BindInstance(
-                    type: valueType,
-                    instance: null,
-                    config: child,
-                    options: options);
-                if (item != null)
+                try
                 {
-                    if (keyType == typeof(string))
-                    {
-                        string key = child.Key;
-                        setter.SetValue(dictionary, item, new object[] { key });
-                    }
-                    else if (keyTypeIsEnum)
+                    object? item = BindInstance(
+                        type: valueType,
+                        instance: null,
+                        config: child,
+                        options: options);
+                    if (item != null)
                     {
-                        object key = Enum.Parse(keyType, child.Key);
-                        setter.SetValue(dictionary, item, new object[] { key });
+                        if (keyType == typeof(string))
+                        {
+                            string key = child.Key;
+                            setter.SetValue(dictionary, item, new object[] { key });
+                        }
+                        else if (keyTypeIsEnum)
+                        {
+                            object key = Enum.Parse(keyType, child.Key);
+                            setter.SetValue(dictionary, item, new object[] { key });
+                        }
                     }
                 }
+                catch
```

Comment:
I'm not sure we should be silently eating all exceptions here. If someone is trying to bind to invalid data, isn't it better that they get an error? That way they know the data was invalid.

Say, for example, someone has a type-o in their config. It would take a long time for them to figure out why the configuration setting isn't applying correctly.

Reply:
Ah, good point. I missed that. Thanks for the clarification.